Sources/StreamChat/Database/DTOs/MessageDTO.swift

```diff
@@ -51,10 +51,15 @@ class MessageDTO: NSManagedObject {
         
         prepareDefaultSortKeyIfNeeded()
     }
-    
+
+    /// We use this value as the offset for the sorting key for unsent messages. Even if the local time of the device
+    /// is different from the server time, we want to be sure the newly sent message is always the last one. By applying
+    /// the same offset to all unsent message, we make sure the order of the messages is maintained.
+    static let localSortingKeyTimeOffset: TimeInterval = 24 * 60 * 60
```

Comment:
I don't think we should do this, in some cases the failed message will make no sense if re-ordered (in the meantime newer messages will be in the channel).

Reply:
Good point 👍 We can either ignore this edge case (all iPhones have the time synced remotely by default - you have to explicitly turn this off and then change the time), or we can just make sure the `locallyCreatedAt` is always bigger than the last message in the channel we locally have. WDYT?